conn.go

```diff
@@ -174,13 +174,19 @@ var (
 		TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
 		TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
 		TLS_PSK_WITH_AES_128_GCM_SHA256,
+		TLS_PSK_WITH_AES_256_GCM_SHA384,
+		TLS_DHE_RSA_WITH_AES_128_GCM_SHA256,
+		TLS_DHE_RSA_WITH_AES_256_GCM_SHA384,
 		TLS_ECDHE_PSK_WITH_AES_128_GCM_SHA256,
+		TLS_ECDHE_PSK_WITH_AES_256_GCM_SHA384,
+		TLS_DHE_PSK_WITH_AES_128_GCM_SHA256,
+		TLS_DHE_PSK_WITH_AES_256_GCM_SHA384,
 	}
 
 	defaultSupportedGroups = []namedGroup{
 		namedGroupP256,
 		namedGroupP384,
-		namedGroupP521,
```

Comment:
Noting for the historical record that this patch is removing P-521 from the default list of groups.  It's unrelated, but I'll let it slide.